### PR TITLE
Mergen team b branch naar develop branch

### PR DESCRIPTION
**Pull request by Joossensei**

### CODE REVIEW

**Review by Chris3007 — APPROVED** [review]
Seems legit

File: lib/Api/apiManager.dart [hunk 1]
```diff
@@ -3,22 +3,24 @@ import '../config.dart';
 import 'package:http/http.dart' as http;
 
 class ApiManager {
-  tryJsonDecode(String data) {
+  static T tryJsonDecode<T>(String data) {
     try {
-      return jsonDecode(data);
+
```
Comment (jimtronchet) [inline lib/Api/apiManager.dart:8]: Witregel?

File: lib/Api/apiManager.dart [hunk 2]
```diff
@@ -3,22 +3,24 @@ import '../config.dart';
...
 class ApiManager {
-  tryJsonDecode(String data) {
+  static T tryJsonDecode<T>(String data) {
     try {
-      return jsonDecode(data);
+      
+      return json.decode(data) as T;
     } catch (error) {
-      throw Exception('Failed to decode json + Reason: $error');
+
```
Comment (jimtronchet) [inline lib/Api/apiManager.dart:11]: Zelfde

File: lib/Stock/stockReport.dart [hunk 3]
```diff
@@ -1,77 +1,102 @@
...
+  Future addProduct(StockProduct stockProduct) async {
+    Map<String, dynamic> body = {
+      'name': stockProduct.product.name,
+      'price': stockProduct.product.price,
+      'icon': stockProduct.product.icon,
+      'color': colorToString(
+          Colors.primaries[Random().nextInt(Colors.primaries.length)])
+    };
+
+    PopupAndLoading.showLoading();
```
Comment (jimtronchet) [inline lib/Stock/stockReport.dart:60]: Nice 👍

**Review by jimtronchet — APPROVED** [review]

**Review by UDefine1 — APPROVED** [review]
oooehh looks niceee i like the way you smile. With your beautifull eyes. oh baby. got to be good, Oh baby you got to be smooth

File: lib/Api/apiManager.dart (hunk 1 above)
Comment (jimtronchet) [reply lib/Api/apiManager.dart:8]: ^

**Review by jimtronchet — CHANGES_REQUESTED** [review]

**Review by jimtronchet — APPROVED** [review]

**Review by wasdafor — APPROVED** [review]
Next time use elements instead of image for the login.

**Pull request merged**